tests/kafkatest/services/kafka/kafka.py

```diff
@@ -410,6 +410,56 @@ def leader(self, topic, partition=0):
         self.logger.info("Leader for topic %s and partition %d is now: %d" % (topic, partition, leader_idx))
         return self.get_node(leader_idx)
 
+    def list_consumer_groups(self, node=None, new_consumer=False, command_config=None):
+        """ Get list of consumer groups.
+        """
+        if node is None:
```

Comment:
Looks like there's a lot of redundant code between `list_consumer_groups` and `describe_consumer_group` since almost all of it is for setting up the `kafka-consumer-groups.sh` command. What about making that a common, fairly generic method and making each of these thin wrappers around it? For example, the generic common version might do all of this initial command setup, just accepting a list of extra argumets (i.e. empty in list_groups, just the group name in describe). Only thing I'm not sure about making generic is the way you're processing the output (which is the unfortunate result of not having something like JSON output for these commands).